openassessment/xblock/submission_mixin.py

```diff
@@ -250,24 +273,36 @@ def save_files_descriptions(self, data, suffix=''):  # pylint: disable=unused-ar
 
         return {'success': True, 'msg': u''}
 
+    def create_team_submission(self, student_sub_data):
+        """ A student submitting for a team should generate matching submissions for every member of the team. """
+        if self.has_team():
+            submitter_anonymous_user_id = self.xmodule_runtime.anonymous_student_id
+            team_anonymous_user_ids = self.get_anonymous_user_ids_for_team()
+
+            submissions = []
+
+            # submissions are identical except for the student item
+            for anonymous_user_id in team_anonymous_user_ids:
+                student_item_dict = self.get_student_item_dict(anonymous_user_id=anonymous_user_id)
+                submission = self.create_submission(student_item_dict, student_sub_data)
+
+                if anonymous_user_id == submitter_anonymous_user_id:
+                    self.submission_uuid = submission["uuid"]
+
+                submissions.append(submission)
+
+        return submissions
```

Comment:
If, somehow, you get here and `self.has_team()` returns false, `return submissions` will result in a `NameError`, because `submissions` is only defined inside the if block.

Reply:
Could we remove the `self.has_team()` check since that's a path that can't be hit currently? Or have an overabundance of caution? Throw an exception? Return `None`?